lib/include/pl/patterns/pattern_pointer.hpp

```diff
@@ -67,18 +69,29 @@ namespace pl {
             this->m_pointedAt->setColor(Pattern::getColor());
         }
 
-        void setPointedAtAddress(u64 address) {
-            this->m_pointedAtAddress = address;
+        void setPointerTypePattern(std::unique_ptr<Pattern> &&pattern) {
+            Pattern::setSize(pattern->getSize());
+            Pattern::setEndian(pattern->getEndian());
+            this->m_pointerType = std::move(pattern);
+            this->m_signed = dynamic_cast<const PatternSigned *>(this->m_pointerType.get()) != nullptr;
         }
 
-        [[nodiscard]] u64 getPointedAtAddress() const {
+        void setPointedAtAddress(i128 address) {
+            this->m_pointedAtAddress = address + this->m_pointerBase;
+        }
+
+        [[nodiscard]] i128 getPointedAtAddress() const {
             return this->m_pointedAtAddress;
         }
 
         [[nodiscard]] const std::shared_ptr<Pattern> &getPointedAtPattern() {
             return this->m_pointedAt;
         }
 
+        [[nodiscard]] bool isSigned() const {
+            return this->isSigned();
```

Comment:
This is an infinite recursion loop. You probably meant to use `this->m_signed` here